### PR TITLE
Document `sriov` interface mode

### PR DESCRIPTION
**Pull request by booxter**
This is to document the new mode added in:
https://github.com/kubevirt/kubevirt/pull/1625

### CODE REVIEW

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 1]
```diff
@@ -223,3 +224,18 @@ At this time, `slirp` mode doesn't support additional configuration fields.
...
+plugin](https://github.com/intel/sriov-network-device-plugin). The device is
+passed through into the guest operating system as a host device, using the
+[vfio](https://www.kernel.org/doc/Documentation/vfio.txt) userspace interface,
+to maintain high networking performance.
+
+> **Note:** while the `sriov` mode is validated and tested using the Intel
+> SR-IOV device plugin, other plugins may add support for the same by setting
+> the `SRIOV-VF-PCI-ADDR` environment variable inside pods to a list of
+> allocated PCI device IDs, as in:
+> SRIOV-VF-PCI-ADDR=0000:81:11.1,0000:81:11.2[,...]
```
Comment (phoracek) [inline workloads/virtual-machines/interfaces-and-networks.md]: please would you add an example with devices-interfaces and networks sections?

Comment (booxter) [reply workloads/virtual-machines/interfaces-and-networks.md]: Thanks, added. Also added a note about limits/requests handling.

**Comment by fabiand** [comment]
Let's hold this back until kubevirt/kubevirt#1625 is solved

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 2]
```diff
@@ -223,3 +224,42 @@ At this time, `slirp` mode doesn't support additional configuration fields.
...
+
+> **Note:** as of the time of writing, in addition to attaching a VM to a
+> SR-IOV network, you should also request corresponding devices from the device
```
Comment (rmohr) [inline workloads/virtual-machines/interfaces-and-networks.md]: Is this more a `should` or a `have to`?

Comment (booxter) [reply workloads/virtual-machines/interfaces-and-networks.md]: You are right.

**Comment by booxter** [comment]
@rmohr @fabiand  can we merge it? We have sriov type in kubevirt master now.

**Pull request merged**